src/Features/CSharp/Portable/ConvertBetweenRegularAndVerbatimString/AbstractConvertBetweenRegularAndVerbatimStringCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,212 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertBetweenRegularAndVerbatimString
+{
+    internal abstract class AbstractConvertBetweenRegularAndVerbatimStringCodeRefactoringProvider<
```

Comment:
It seems like an overkill to include the name of the refactoring in both the containing namespace and the typename.

Reply:
> It seems like an overkill to include the name of the refactoring in both the containing namespace and the typename.

This is how we name all our refactorings/analyzers/fixers.

Reply:
> I'd expect us to have all these under a common namespace 

We don't do that.  We might make that happen when/if we move this to lower layers or into a dll dedicated to refactorings.